src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +973,73 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
             return not time_is_expired if check_node_is_valid else time_is_expired
         return False
 
+    def _handle_ice_nodes(self, active_nodes, unhealthy_dynamic_nodes):
+        ice_nodes = [node for node in unhealthy_dynamic_nodes if node.is_ice()]
```

Comment:
I think I saw this somewhere else. We could have a function that takes as input a list of nodes and returned ice nodes (e.g. `get_unhealthy_ice_nodes`). Up to you to decide where to put it. One possibility is in the same module where Nodes classes are defined. it could even be a static method of the SlurmNode class.